test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/EntityTypeTest.cs

```diff
@@ -692,6 +692,48 @@ public void GetEdmModel_WorksOnModelBuilder_ForOpenEntityType()
             Assert.True(entityType.Properties().Where(c => c.Name == "Name").Any());
         }
 
+
+
+        [Fact]
+        public void DynamicDictionaryProperty_Works_ToSetEntityTypeWithAnnotation()
+        {
+            // Arrange
+            ODataModelBuilder builder = new ODataModelBuilder();
+
+            // Act
+            EntityTypeConfiguration<SimpleAnnotationEntityType> entityType = builder.EntityType<SimpleAnnotationEntityType>();
+            entityType.HasKey(c => c.Id);
+            entityType.Property(c => c.Name);
+            entityType.HasInstanceAnnotations(c => c.InstanceAnnotations);
+
+            // Assert
+            Assert.True(entityType.Configuration.InstanceAnnotationsDictionary!=null);
+        }
+
+        [Fact]
+        public void GetEdmModel_WorksOnModelBuilder_ForAnnotationEntityType()
+        {
+            // Arrange
+            ODataModelBuilder builder = new ODataModelBuilder();
+            EntityTypeConfiguration<SimpleAnnotationEntityType> entity = builder.EntityType<SimpleAnnotationEntityType>();
+            entity.HasKey(c => c.Id);
+            entity.Property(c => c.Name);
+            entity.HasInstanceAnnotations(c => c.InstanceAnnotations);
+
+            // Act
+            IEdmModel model = builder.GetEdmModel();
+
+            // Assert
+            Assert.NotNull(model);
+            IEdmEntityType entityType = Assert.Single(model.SchemaElements.OfType<IEdmEntityType>());
+            
+            Assert.Equal(2, entityType.Properties().Count());
+
+            Assert.True(entityType.Properties().Where(c => c.Name == "Id").Any());
```

Comment:
>Assert.True(entityType.Properties().Where(c => c.Name == "Id").Any()); [](start = 12, length = 70)

Call "GetInstantceAnnotations" from the model to check the annotation is set on the model?